From 8f956e7fe22e99597b759845083ac784d15ed0dc Mon Sep 17 00:00:00 2001
From: khaleel <khaleel.s@phytec.in>
Date: Fri, 23 Dec 2022 15:53:45 +0530
Subject: [PATCH 2/6] Device tree support for RB-imx6ul rdk

Configures all the required peripherals and interfaces
for RB-imx6ul rdk board

Signed-off-by: khaleel <khaleel.s@phytec.in>
---
 .../arm/boot/dts/imx6ul-phytec-ruggedboard-rdk.dts |  83 +++++
 arch/arm/boot/dts/imx6ul-phytec-ruggedboard.dtsi   | 376 +++++++++++++++++++++
 2 files changed, 459 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6ul-phytec-ruggedboard-rdk.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-phytec-ruggedboard.dtsi

diff --git a/arch/arm/boot/dts/imx6ul-phytec-ruggedboard-rdk.dts b/arch/arm/boot/dts/imx6ul-phytec-ruggedboard-rdk.dts
new file mode 100644
index 00000000..d3d5824
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-phytec-ruggedboard-rdk.dts
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2020 PHYTEC INDIA PVT LTD
+ * Author: Sai kiran M <kiran.m@phytec.in>
+ */
+
+/dts-v1/;
+#include "imx6ul.dtsi"
+#include "imx6ul-phytec-phycore-som.dtsi"
+#include "imx6ul-phytec-ruggedboard.dtsi"
+#include "imx6ul-phytec-segin-peb-av-02.dtsi"
+
+/ {
+
+	model = "Rugged Board i.MX6 UltraLite";
+	compatible = "phytec,imx6ul-pbacd10", "phytec,imx6ul-pcl063","fsl,imx6ul";
+};
+
+&adc1 {
+	status = "okay";
+};
+
+&can1 {
+	status = "okay";
+};
+
+&ethphy1 {
+	status = "okay";
+};
+
+&fec1 {
+	status = "okay";
+};
+
+&fec2 {
+	status = "disabled";
+};
+
+&gpmi {
+	status = "okay";
+};
+
+&reg_can1_en {
+	status = "okay";
+};
+
+&uart2 {
+	status = "okay";
+};
+
+&uart3 {
+	status = "okay";
+};
+
+&uart5 {
+	status = "okay";
+};
+
+&uart6 {
+	rs485-rts-active-high;
+	linux,rs485-enabled-at-boot-time;
+	status = "okay";
+};
+
+&usbotg1 {
+	status = "okay";
+};
+
+&usbotg2 {
+	status = "okay";
+};
+
+&usdhc1 {
+	status = "okay";
+};
+
+&usdhc2 {
+	status = "okay";
+};
+
+&pwm3 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6ul-phytec-ruggedboard.dtsi b/arch/arm/boot/dts/imx6ul-phytec-ruggedboard.dtsi
new file mode 100644
index 00000000..cbb38f7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-phytec-ruggedboard.dtsi
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2020 PHYTEC INDIA PVT LTD
+ * Author: Sai kiran M <kiran.m@phytec.in>
+ */
+
+/ {
+	model = "Rugged Board i.MX6 UltraLite";
+	compatible = "phytec,imx6ul-pbacd-10", "phytec,imx6ul-pcl063","fsl,imx6ul";
+
+	aliases {
+		rtc0 = &i2c2_rtc;   /* RV-4162-C7 RTC */
+	};
+
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_can1_en: regulator@3 {
+			compatible = "regulator-fixed";
+			reg = <3>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&princtrl_flexcan1_en>;
+			regulator-name = "Can";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio5 2 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+			status = "disabled";
+		};
+
+		reg_adc1_vref_3v3: regulator@5 {
+			compatible = "regulator-fixed";
+			regulator-name = "vref-3v3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+		};
+
+		reg_wlan_en: regulator@7 {
+			compatible = "regulator-fixed";
+			reg = <7>;
+			regulator-name = "wlan_en";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio1 18 GPIO_ACTIVE_HIGH>;
+			startup-delay-us = <70000>; /* card specific delay */
+			enable-active-high;
+			status = "okay";
+		};
+
+	};
+
+};
+
+&can1 {							/* CAN TX & RX on P9 Connector(PIN-1,2) */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1>;
+	xceiver-supply = <&reg_can1_en>;
+	status = "disabled";
+};
+
+&i2c2 {
+	pinctrl-names = "default";
+	pinctrl-0 =<&pinctrl_i2c2>;
+	clock-frequency = <100000>;
+	status = "okay";
+
+	i2c2_rtc: rtc@68 {				/* RTC */
+		compatible = "rv4162";
+		reg = <0x68>;
+		status = "okay";
+	};
+};
+
+&pwm3 {							/* PWM3 Controller for GPIO1_4 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm3>;
+	status = "disabled";
+};
+
+&uart2{							/* UART2 full-Duplex communication on P4 connector */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "disabled";
+
+};
+
+&uart3 {						/* UART 3 Half-Duplex communication on Expansion Connector */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3>;
+	status = "disabled";
+};
+
+&uart5 {						/* UART 5 Full-duplex Communication for BLE or RX can be used as USB OTG2 ID PIN */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart5>;
+	uart-has-rtscts;
+	status = "disabled";
+};
+
+&uart6 {						/* UART 6 HAlf Duplex Communication for RS-485 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6>;
+	rts-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+	rs485-rts-delay = <0 0>;
+	status = "disabled";
+};
+
+&usbotg1{						/* USB 0 ID Pin */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb_otg1_id>;
+	dr_mode = "host";
+	disable-over-current;
+	status = "disabled";
+};
+
+&usbotg2 {						/* USB 1 ID Pin */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb_otg2_id>;
+	dr_mode = "host";
+	disable-over-current;
+	status = "disabled";
+};
+
+&usdhc1 {						/* SD-CARD mmc lines */
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
+	cd-gpios = <&gpio1 19 GPIO_ACTIVE_LOW>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	wakeup-source;
+	status = "disabled";
+};
+
+&usdhc2 {						/* ATWILC3000 mmc communication lines */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2_wlan &pinctrl_wlan>;
+	vmmc-supply = <&reg_wlan_en>;
+	bus-width = <4>;
+	non-removable;
+	cap-power-off-card;
+	keep-power-in-suspend;
+	status = "disabled";
+};
+
+&adc1 {							/* ADC Pin on the micro bus GPIO1_3 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_adc1>;
+	vref-supply = <&reg_adc1_vref_3v3>;
+	status = "disabled";
+	/*
+	 * driver can not separate a specific channel so we request 4 channels
+	 * here - we need only the fourth channel
+	 */
+	num-channels = <4>;
+};
+
+&ecspi2 {						/* SPI1 Communication on the Expansion */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi2>;
+	cs-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
+	dmas = <&sdma 5 7 1>,
+	       <&sdma 6 7 2>;
+	dma-names = "rx", "tx";
+	status = "okay";
+
+	spidev1: spi@0 {
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+		status = "okay";
+	};
+};
+
+&ecspi1{							/* SPI0 Communication on the Expansion */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	cs-gpios = <&gpio4 26 GPIO_ACTIVE_LOW>;
+	dmas = <&sdma 3 7 1>,
+		<&sdma 4 7 2>;
+	dma-names = "rx", "tx";
+	status = "okay";
+
+	spidev0: spi@0 {
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+		status = "okay";
+	};
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpios>;
+	pinctrl_gpios: gpios {
+		fsl,pins = <
+			MX6UL_PAD_ENET2_TX_DATA0__GPIO2_IO11	0x17059     /* User LED-1 */
+			MX6UL_PAD_ENET2_TX_DATA1__GPIO2_IO12	0x17059	    /* User LED-2 */
+			MX6UL_PAD_ENET2_TX_EN__GPIO2_IO13	0x17059     /* Digital input */
+			MX6UL_PAD_ENET2_RX_DATA0__GPIO2_IO08	0x17059     /* User Switch */
+			MX6UL_PAD_SNVS_TAMPER5__GPIO5_IO05	0x17059     /* Digital input -1 */
+			MX6UL_PAD_GPIO1_IO01__GPIO1_IO01	0x17059     /* Digital input -3 */
+			MX6UL_PAD_JTAG_TMS__GPIO1_IO11		0x4001b0b0  /* Digital input -4 or Expansion GPIO(31) */
+			MX6UL_PAD_JTAG_TDI__GPIO1_IO13          0x4001b0b0  /* Digital output -1 or Expansion GPIO(5) */
+			MX6UL_PAD_JTAG_TDO__GPIO1_IO12		0x4001b0b0  /* Digital output -2 or Expansion GPIO(7) */
+			MX6UL_PAD_JTAG_TRST_B__GPIO1_IO15	0x4001b0b0  /* Digital output -3 or Expansion GPIO(9) */
+			MX6UL_PAD_JTAG_TCK__GPIO1_IO14		0x4001b0b0  /* Digital output -4 or Expansion GPIO(11) */
+			MX6UL_PAD_ENET2_RX_ER__GPIO2_IO15	0x17059     /* USB Control GPIO */
+			MX6UL_PAD_ENET2_RX_DATA1__GPIO2_IO09	0x17059     /* USB Control GPIO */
+			MX6UL_PAD_ENET2_TX_CLK__GPIO2_IO14	0x17059     /* Expansion GPIO(29) */
+			MX6UL_PAD_SNVS_TAMPER0__GPIO5_IO00	0x17059     /* Micro Bus Interrupt */
+			MX6UL_PAD_JTAG_MOD__GPIO1_IO10          0x4001b0b0  /* Expansion GPIO(35) */
+		>;
+	};
+
+	pinctrl_flexcan1: flexcan1 {
+		fsl,pins = <
+			MX6UL_PAD_UART3_CTS_B__FLEXCAN1_TX	0x0b0b0
+			MX6UL_PAD_UART3_RTS_B__FLEXCAN1_RX	0x0b0b0
+		>;
+	};
+
+	princtrl_flexcan1_en: flexcan1engrp {
+		fsl,pins = <
+			MX6UL_PAD_SNVS_TAMPER2__GPIO5_IO02	0x17059
+		>;
+	};
+
+	pinctrl_i2c2: i2c2grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_HSYNC__I2C2_SCL      0x4001b8b0
+			MX6UL_PAD_CSI_VSYNC__I2C2_SDA      0x4001b8b0
+		>;
+	};
+
+	pinctrl_pwm3: pwm3grp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO04__PWM3_OUT		0x1b0b1
+		>;
+	};
+
+	pinctrl_rtc_int: rtcintgrp {
+		fsl,pins = <
+			MX6UL_PAD_SNVS_TAMPER1__GPIO5_IO01	0x17059
+		>;
+	};
+
+	pinctrl_stmpe: stmpegrp {
+		fsl,pins = <
+			MX6UL_PAD_SNVS_TAMPER3__GPIO5_IO03	0x17059
+		>;
+	};
+
+	pinctrl_uart2: uart2grp {
+		fsl,pins = <
+			MX6UL_PAD_UART2_TX_DATA__UART2_DCE_TX   0x1b0b1
+			MX6UL_PAD_UART2_RX_DATA__UART2_DCE_RX   0x1b0b1
+			MX6UL_PAD_UART2_CTS_B__UART2_DCE_CTS    0x1b0b1
+			MX6UL_PAD_UART2_RTS_B__UART2_DCE_RTS    0x1b0b1
+		>;
+	};
+
+	pinctrl_uart3: uart3grp {
+		fsl,pins = <
+			MX6UL_PAD_UART3_RX_DATA__UART3_DCE_RX	0x1b0b1
+			MX6UL_PAD_UART3_TX_DATA__UART3_DCE_TX	0x1b0b1
+		>;
+	};
+
+	pinctrl_uart5: uart5grp {
+		fsl,pins = <
+			MX6UL_PAD_UART5_TX_DATA__UART5_DCE_TX	0x1b0b1
+			MX6UL_PAD_UART5_RX_DATA__UART5_DCE_RX	0x1b0b1
+			MX6UL_PAD_GPIO1_IO08__UART5_DCE_RTS	0x1b0b1
+			MX6UL_PAD_GPIO1_IO09__UART5_DCE_CTS	0x1b0b1
+		>;
+	};
+
+	pinctrl_uart6: uart6grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_MCLK__UART6_DCE_TX	0x0b0b0
+			MX6UL_PAD_CSI_PIXCLK__UART6_DCE_RX      0x0b0b0
+			MX6UL_PAD_ENET2_RX_EN__GPIO2_IO10       0x0b0b0
+		>;
+	};
+
+	pinctrl_usb_otg1_id: usbotg1idgrp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO00__ANATOP_OTG1_ID	0x17059
+		>;
+	};
+
+	pinctrl_usb_otg2_id: usbotg2idgrp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO05__ANATOP_OTG2_ID	0x17059
+		>;
+	};
+
+	pinctrl_usdhc1: usdhc1grp {
+		fsl,pins = <
+			MX6UL_PAD_SD1_CMD__USDHC1_CMD		0x17059
+			MX6UL_PAD_SD1_CLK__USDHC1_CLK		0x10059
+			MX6UL_PAD_SD1_DATA0__USDHC1_DATA0	0x17059
+			MX6UL_PAD_SD1_DATA1__USDHC1_DATA1	0x17059
+			MX6UL_PAD_SD1_DATA2__USDHC1_DATA2	0x17059
+			MX6UL_PAD_SD1_DATA3__USDHC1_DATA3	0x17059
+			MX6UL_PAD_UART1_RTS_B__GPIO1_IO19	0x17059
+		>;
+	};
+
+	pinctrl_usdhc1_100mhz: usdhc1grp100mhz {
+		fsl,pins = <
+			MX6UL_PAD_SD1_CMD__USDHC1_CMD		0x170b9
+			MX6UL_PAD_SD1_CLK__USDHC1_CLK		0x100b9
+			MX6UL_PAD_SD1_DATA0__USDHC1_DATA0	0x170b9
+			MX6UL_PAD_SD1_DATA1__USDHC1_DATA1	0x170b9
+			MX6UL_PAD_SD1_DATA2__USDHC1_DATA2	0x170b9
+			MX6UL_PAD_SD1_DATA3__USDHC1_DATA3	0x170b9
+		>;
+	};
+
+	pinctrl_usdhc1_200mhz: usdhc1grp200mhz {
+		fsl,pins = <
+			MX6UL_PAD_SD1_CMD__USDHC1_CMD		0x170f9
+			MX6UL_PAD_SD1_CLK__USDHC1_CLK		0x100f9
+			MX6UL_PAD_SD1_DATA0__USDHC1_DATA0	0x170f9
+			MX6UL_PAD_SD1_DATA1__USDHC1_DATA1	0x170f9
+			MX6UL_PAD_SD1_DATA2__USDHC1_DATA2	0x170f9
+			MX6UL_PAD_SD1_DATA3__USDHC1_DATA3	0x170f9
+		>;
+	};
+
+	pinctrl_usdhc2_wlan: usdhc2grp_wlan {
+		fsl,pins = <
+			MX6UL_PAD_LCD_DATA18__USDHC2_CMD        0x17059
+			MX6UL_PAD_LCD_DATA19__USDHC2_CLK        0x10059
+			MX6UL_PAD_LCD_DATA20__USDHC2_DATA0      0x17059
+			MX6UL_PAD_LCD_DATA21__USDHC2_DATA1      0x17059
+			MX6UL_PAD_LCD_DATA22__USDHC2_DATA2      0x17059
+			MX6UL_PAD_LCD_DATA23__USDHC2_DATA3      0x17059
+		>;
+	};
+
+	pinctrl_wlan: wlangrp {
+		fsl,pins = <
+			MX6UL_PAD_NAND_DQS__GPIO4_IO16		 0x80000000    /* wilc irq line */
+		>;
+	};
+
+	pinctrl_adc1: adc1grp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO03__GPIO1_IO03	0xb0
+		>;
+	};
+
+	pinctrl_ecspi1: ecspi1grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_DATA07__ECSPI1_MISO    0x170f9
+			MX6UL_PAD_CSI_DATA06__ECSPI1_MOSI    0x170f9
+			MX6UL_PAD_CSI_DATA04__ECSPI1_SCLK    0x170f9
+			MX6UL_PAD_CSI_DATA05__GPIO4_IO26     0x170f9
+		>;
+	};
+
+	pinctrl_ecspi2: ecspi2grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_DATA03__ECSPI2_MISO    0x170f9
+			MX6UL_PAD_CSI_DATA02__ECSPI2_MOSI    0x170f9
+			MX6UL_PAD_CSI_DATA01__GPIO4_IO22     0x170f9
+			MX6UL_PAD_CSI_DATA00__ECSPI2_SCLK    0x170f9
+		>;
+	};
+};
-- 
2.7.4

